### PR TITLE
Features dev

### PR DESCRIPTION
**Pull request by BessemMnif**
- Add links by wizard
- visualizing links in both graph and list format
- Handling multiple things description

### CODE REVIEW

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+     const context = useContext(ediTDorContext);
+     const [display, setDisplay] = React.useState(() => { return false });
+     const [linkingMethod, setlinkingMethod] = React.useState(() => { return "url" });
+     const [currentLinkedTd, setCurrentLinkedTd] = React.useState(() => { return {} });
+
+
+     const interaction = props.interaction ?? {};
+     const offlineTD = JSON.parse(context.offlineTD)
+
+     useEffect(() => {
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:29]: Do we need the useEffect() if there is nothing done in?

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+
+     const checkIfLinkExists = (link) => {
+         if (hasLinks(interaction)) {
+            return checkIfLinkIsInItem(link, interaction)
+         }
+         return false
+     }
+
+     const addLinksToInteraction = (link) => {
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:55]: Are the Links really added to an interaction?

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                     td = JSON.parse(context.offlineTD);
+                 } catch (_) { }
+                 if (!hasLinks(td)) {
+                     td.links = [];
+                 }
+                 td.links.push(link);
+                 context.updateOfflineTD(JSON.stringify(td, null, 2))
+     }
+
+     const getFileHandle = () => {
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:67]: Isn`t that method already in AppHeader.jsx? 
Maybe a good candidate for moving to util.js in order to have duplicated code for opening files?

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+            },
+          ],
+        };
+        if ("showOpenFilePicker" in window) {
+          return window.showOpenFilePicker(opts).then((handles) => handles[0]);
+        }
+        return window.chooseFileSystemEntries();
+      };
+
+     const getFileHTML5 = async () => {
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:82]: Isn`t that method already in AppHeader.jsx?
Maybe a good candidate for moving to util.js in order to have duplicated code for opening files?

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+              return resolve(file);
+            }
+            return reject(new Error("AbortError"));
+          };
+          fileInput.click();
+        });
+      };
+
+
+     const hasNativeFS = useCallback(() => {
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:97]: Isn`t that method already in AppHeader.jsx?
Maybe a good candidate for moving to util.js in order to have duplicated code for opening files?

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+          "chooseFileSystemEntries" in window || "showOpenFilePicker" in window
+        );
+      }, []);
+
+      /**
+       *
+       * @param {*} file
+       *
+       * Read file content
+       */
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:108]: This comment is a bit confusing here in my opinion.

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+             <input
+                list="relationType"
+                 type="text"
+                 name="rel"
+                 id="rel"
+                 className="border-gray-600 bg-gray-600 w-full p-2 sm:text-sm border-2 text-white rounded-md focus:outline-none focus:border-blue-500"
+                 placeholder="relation name"
+             />
+              <datalist id="relationType">
+                <option value="icon" />
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:161]: Can we have these options as array? 
Maybe a good idea could be to get them from the JSONschema or somewhere directly?

File: src/components/Dialogs/AddLinkTdDialog.jsx [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+                <option value="item" />
+                <option value="predecessor-version" />
+                <option value="controlledBy" />
+             </datalist>
+
+             <span id="link-rel-info" className="text-xs text-red-400 pl-2"></span>
+         </div>
+         <div className="p-1 pt-2">
+             <label htmlFor="link-href" className="text-sm text-gray-400 font-medium pl-2" >Target ressource:</label>
+                &nbsp;&nbsp;&nbsp;&nbsp;
```
Comment (fexpal) [inline src/components/Dialogs/AddLinkTdDialog.jsx:177]: Do we need the `&nbsp;` here?

File: src/util.js [hunk 9]
```diff
@@ -18,7 +18,12 @@
  * 
  * @description
  * Parses all key-value pairs of an object into an object'. 
+ * 
  */
+
```
Comment (fexpal) [inline src/util.js:23]: Please delete the empty lines.

File: src/components/Dialogs/AddLinkTdDialog.jsx (hunk 5 above)
Comment (BessemMnif) [reply src/components/Dialogs/AddLinkTdDialog.jsx:97]: hasNativeFs use a Hook (useCallback) and can only be used inside a functional component. Cannot be moved to util.js

File: src/components/App/App.css [hunk 10]
```diff
@@ -65,4 +65,19 @@ details[open]>summary::before {
...
   display: none;
+}
+#tabsBackground{
+  background-color: #1E1E1E;
+  height: 25px;
+}
+#linkedTd{
+  background-color: #1E1E1E ;
+  color: white;
+  width: calc( 100vh - 500px );
```
Comment (fexpal) [inline src/components/App/App.css:76]: I think pixel values are not really responsive.

File: src/components/App/App.css [hunk 11]
```diff
@@ -65,4 +65,19 @@ details[open]>summary::before {
...
 details>summary::marker {
   display: none;
+}
+#tabsBackground{
+  background-color: #1E1E1E;
+  height: 25px;
+}
+#linkedTd{
+  background-color: #1E1E1E ;
+  color: white;
```
Comment (fexpal) [inline src/components/App/App.css:75]: What about https://tailwindcss.com/docs/text-color

**Pull request merged**